examples/pet-mad/pet-mad.py

```diff
@@ -0,0 +1,529 @@
+tasks, in `this preprint <http://arxiv.org/TBD>`_.
+"""
+
+# %%
+#
+# Start by importing the required libraries.
+
+import os
+import subprocess
+from copy import copy, deepcopy
+
+# ASE and i-PI scripting utilities
+import ase.units
+import chemiscope
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+from ase.optimize import LBFGS
+from ipi.utils.mathtools import get_rotation_quadrature_lebedev
+from ipi.utils.parsing import read_output, read_trajectory
+from ipi.utils.scripting import (
+    InteractiveSimulation,
+    forcefield_xml,
+    motion_nvt_xml,
+    simulation_xml,
+)
+
+# metatensor models
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+from metatrain.utils.io import load_model
+
+
+if hasattr(__import__("builtins"), "get_ipython"):
+    get_ipython().run_line_magic("matplotlib", "inline")  # noqa: F821
+
+
+# %%
+# Inference on the MAD test set
+# -----------------------------
+#
+# We begin by using the ``ase``-compatible calculator to evaluate
+# energy and forces for a test dataset that contains both hold-out
+# structures from the MAD dataset, and a few structures from popular
+# datasets (MPtrj, Alexandria, OC2020, SPICE, MD22) re-computed with
+# consistent DFT settings.
+#
+# Load the dataset
+# ^^^^^^^^^^^^^^^^
+#
+# We fetch the dataset, and load only some of the structures, to
+# speed up the example runtime on CPU. The model can also run (much faster)
+# on GPUs if you have some at hand.
+
+filename = "data/mad-test-mad-settings.xyz"
+if not os.path.exists(filename):
+    url = (
+        "https://huggingface.co/lab-cosmo/pet-mad/resolve/"
+        "main/benchmarks/mad-test-mad-settings-v1.0.xyz"
+    )
+    response = requests.get(url)
+    response.raise_for_status()
+    with open(filename, "wb") as f:
+        f.write(response.content)
+
+test_structures = ase.io.read(filename, "::16")
+
+# also extract reference energetics and metadata
+test_energy = []
+test_forces = []
+test_natoms = []
+test_origin = []
+subsets = []
+for s in test_structures:
+    test_energy.append(s.get_potential_energy())
+    test_natoms.append(len(s))
+    test_forces.append(s.get_forces())
+    test_origin.append(s.info["origin"])
+    if not s.info["origin"] in subsets:
+        subsets.append(s.info["origin"])
+test_natoms = np.array(test_natoms)
+test_origin = np.array(test_origin)
+test_energy = np.array(test_energy)
+test_forces = np.array(test_forces, dtype=object)
+
+
+# %%
+#
+# Load the model
+# ^^^^^^^^^^^^^^
+#
+# To start using PET-MAD, we first have to load the model. We will use the latest
+# version of the model. PET-MAD is distributed as a check-point file, that
+# allows re-training and fine-tuning, and needs to be *exported* to be used in
+# calculators and to make inference.
+
+mad_huggingface = (
+    "https://huggingface.co/lab-cosmo/pet-mad/resolve/main/models/pet-mad-latest.ckpt"
+)
+model = load_model(mad_huggingface).export()
```

Comment:
We need to be careful: this doesn't torchscript, potentially sacrificing some speed in ASE, which is what people will look for immediately.

The loss in speed can be avoided by saving and reloading, in two ways:
- either `model.save("model.pt", collect_extensions="extensions")`
- or `mtt export https://huggingface.co/lab-cosmo/pet-mad/resolve/main/models/pet-mad-latest.ckpt` from the command line

Then the ASE calculator below needs to be initialized with
`calculator = MetatensorCalculator("model.pt", device="cpu", extensions_dir="extensions")`

Reply:
Right, makes sense.

Reply:
OK addressed this and it is also a better point to discuss exporting. I also explained a bit how ckpt and pt differ.